### PR TITLE
Initializing a size_t with a constant double gives compilation error …

### PR DESCRIPTION
**Pull request by gerk59**
…on VS2017 (32 bits). Changed constant to 0.

### CODE REVIEW

File: include/highfive/bits/H5DataType_misc.hpp [hunk 1]
```diff
@@ -120,18 +120,18 @@ inline AtomicType<std::string>::AtomicType() {
 
 
 
-template <> 
+template <>
 inline AtomicType<std::complex<double> >::AtomicType()
```
[the remark this answers is not in the document]
Comment (gerk59) [reply include/highfive/bits/H5DataType_misc.hpp]: hi Hernando, I didn't look at the actual implementation, only reported a compilation error (initializing an offset with a double).
Do you plan to commit your changes?

[the remark this answers is not in the document]
Comment (hernando) [reply include/highfive/bits/H5DataType_misc.hpp]: I would, but I didn't write the original code, so I'd wait until @adevress gives his opinion, maybe there's an obscure reason for the original implementation. @gerk59, Could you confirm if the code above compiles in VS2017?

[the remark this answers is not in the document]
Comment (gerk59) [reply include/highfive/bits/H5DataType_misc.hpp]: hi Hernando, your suggestion compiles, but I cannot test it on Win32 platform (VS2017 - 32 bits codegen):
there are other places in the code where 32 bit fails (mainly returning a hsize_t to size_t or haddr_t to size_t), and by default warning is treated as error (at least in our project where we're about to embed HighFive).

Would it be wise to add a 32 bit configuration to the tests (or is that already present) ?

Severity	Code	Description	Project	File	Line	Suppression State
Error	C2668	'HighFive::details::vectors_to_single_buffer': ambiguous call to overloaded function	select_partial_dataset_cpp11_bin	c:\proef\highfive.gerk59\include\highfive\bits\h5converter_misc.hpp	69	
Warning	C4244	'return': conversion from 'haddr_t' to 'size_t', possible loss of data	select_partial_dataset_cpp11_bin	c:\proef\highfive.gerk59\include\highfive\bits\h5dataset_misc.hpp	63	
Warning	C4244	'return': conversion from 'hsize_t' to 'size_t', possible loss of data	select_partial_dataset_cpp11_bin	c:\proef\highfive.gerk59\include\highfive\bits\h5attribute_misc.hpp	37	
Warning	C4244	'return': conversion from 'hsize_t' to 'size_t', possible loss of data	select_partial_dataset_cpp11_bin	c:\proef\highfive.gerk59\include\highfive\bits\h5dataset_misc.hpp	37	
Warning	C4309	'static_cast': truncation of constant value	select_partial_dataset_cpp11_bin	c:\proef\highfive.gerk59\include\highfive\bits\h5dataspace_misc.hpp	122

[the remark this answers is not in the document]
Comment (hernando) [reply include/highfive/bits/H5DataType_misc.hpp]: I'm afraid that we don't have the bandwidth to fix the support for 32bit architectures, but fell free to submit a PR for it.

Regarding this PR I'm closing it in favor of #128, which I've just created.

**Comment by hernando** [comment]
Addressed in #128

**Pull request closed without merging**